src/kontrol/solc_to_k.py

```diff
@@ -138,16 +163,74 @@ def flattened(self) -> list[Input]:
             return [self]
 
 
-def inputs_from_abi(abi_inputs: Iterable[dict]) -> list[Input]:
+def inputs_from_abi(abi_inputs: Iterable[dict], natspec_lengths: dict | None) -> list[Input]:
     inputs = []
-    i = 0
+    index = 0
     for input in abi_inputs:
-        cur_input = Input.from_dict(input, i)
+        cur_input = Input.from_dict(input, index, natspec_lengths)
         inputs.append(cur_input)
-        i += len(cur_input.flattened())
+        index += len(cur_input.flattened())
     return inputs
 
 
+def get_input_length(input_dict: dict, lengths: dict) -> tuple[list[int] | None, int | None]:
```

Comment:
In future, we will want to have a function along the lines of `get_annotations` - is the logic extensible?

Reply:
Renaming this to `process_length_equals`.